prince-of-versions/src/test/java/co/infinum/princeofversions/PrinceOfVersionsTest.java

```diff
@@ -881,6 +881,25 @@ public void testCheckingOptionalUpdateWithSdkValuesAndIncreaseInMinor() {
         verify(callback, times(1)).onError(anyThrowable());
     }
 
+    @Test
+    public void testCheckingForUpdateWhenRequirementsAreNull() {
+        Storage storage = new MockStorage();
+        MockApplicationConfiguration applicationConfiguration = new MockApplicationConfiguration(230,
+            16);
+        MockDefaultRequirementChecker checker = new MockDefaultRequirementChecker(applicationConfiguration);
+
+        PrinceOfVersions princeOfVersions = new PrinceOfVersions(storage, new SingleThreadExecutor(), applicationConfiguration,
+            Collections.<String, RequirementChecker>singletonMap(DEFAULT_OS_VERSION, checker));
+        princeOfVersions.checkForUpdatesInternal(
+            new SingleThreadExecutor(),
+            new ResourceFileLoader("update_with_requirements_null.json"),
+            callback
+        );
+
+        verify(callback, times(0)).onSuccess(any(UpdateResult.class));
+        verify(callback, times(1)).onError(anyThrowable());
```

Comment:
Why does it report an error? It should be valid to have null requirement, it would just be skipped. Do you agree?

Reply:
As mentioned above, the `areRequirementsSatisfied` of `PrinceOfVersionsRequirementsProcessor` uses the `checkRequirements` which throws an error since it can't parse null value for requirements (it expects an integer) which in turn returns false for the whole update. 
As far as I understood, that part should not be touched. If I'm mistaken or that needs changing let me know and I'll get on it 👍

Reply:
But there is no requirements for this build and in that case none of the checkers should be checked. If you set some checker to null that basically mean you don't want to check it, you want it to behave as default.
Does that make sense to you?

Reply:
My idea is that having no requirements specified should behave the same as having requirements with null values

Reply:
I understand completely 👍  I'll make the necessary edits